### PR TITLE
upgrade-2.x: redo supported device slug test

### PR DESCRIPTION
**Pull request by imrehg**
The test previously conflated two different things: device type support
and setting a variable for device architecture that is needed in one of
the hotpatch steps. Now disentangle these:

* make the hotpatch-related architecture check automatic, and do it when   it is needed
* only do device type support check if the update version is not to a   hostapp-enabled version
* make it clearer that the device support check is needed to know how   to switch the boot partition on those OS versions (when it's not   the hostapp update hooks that do that)
* deprecate the `--assume-supported` flag, as it has no meaning now, practically

Change-type: minor
Signed-off-by: Gergely Imreh <gergely@balena.io>

### CODE REVIEW

File: upgrade-2.x.sh [hunk 1]
```diff
@@ -815,16 +815,18 @@ fi
...
         binary_type=arm
         ;;
     intel-edison|intel-nuc|iot2000|up-board|qemux86*)
         binary_type=x86
         ;;
     *)
-        if { version_gt "$target_version" "$minimum_hostapp_target_version" || [ "$target_version" == "$minimum_hostapp_target_version" ]; } &&
+        if version_gt "${VERSION_ID}" "${minimum_hostapp_target_version}" || [ "${VERSION_ID}" == "${minimum_hostapp_target_version}" ] ; then
+            log "Update implicitly supported with hostapp-based updates for ${SLUG} device type."
+        elif { version_gt "$target_version" "$minimum_hostapp_target_version" || [ "$target_version" == "$minimum_hostapp_target_version" ]; } &&
```
Comment (xginn8) [inline upgrade-2.x.sh]: small nit: can we style the variables consistently here?
```suggestion
        elif { version_gt "${target_version}" "${minimum_hostapp_target_version}" || [ "${target_version}" == "${minimum_hostapp_target_version}" ]; } &&
```

Comment (imrehg) [reply upgrade-2.x.sh]: @xginn8 fair, should have done it already... Will do that. 
Also, these scripts would need some bit of housekeeping (of not real changes to functionality, just mostly cleaning up - hopefully some day soon)

Comment (imrehg) [reply upgrade-2.x.sh]: @xginn8 pushed changes as per your uggestion

**Comment by imrehg** [comment]
@hedss no, devices show up as as `i586` (IOT2000)/`i686`(Edison), nothing below that. `uname -i` if works on the device, then it would give `i386` (not `uname -m` which gives these results). But repushed with a wildcard check of that for you :)

**Pull request merged**